### PR TITLE
feat(ui): shell completion for envs and apps

### PR DESCRIPTION
**Pull request by fritzduchardt**
…rovide to "envs/" prefix

### CODE REVIEW

File: internal/myks/globe.go [hunk 1]
```diff
@@ -398,3 +398,22 @@ func (g *Globe) getSyncTools() []SyncTool {
...
+	for envPath, val := range envSearchPathToAppMap {
+		envAppMap[g.AddBaseDirToEnvPath(envPath)] = val
+	}
+	return envAppMap
+}
+
+func (g *Globe) AddBaseDirToEnvPath(envName string) string {
+	if strings.HasPrefix(envName, g.EnvironmentBaseDir) {
+		return envName
+	}
```
Comment (Zebradil) [inline internal/myks/globe.go:417]: This should ensure that we correctly process environment names that start with the same substring as the base dir (e.g. `envsio`, `envstarter`, etc.). 

```suggestion
	if strings.HasPrefix(envName, g.EnvironmentBaseDir+filepath.Separator) {
		return envName
	}
```

(This is not tested)

**Review by Zebradil — CHANGES_REQUESTED** [review]
I left a couple of suggestions.

Otherwise, LGTM. Thanks for implementing this!

File: internal/myks/globe.go [hunk 2]
```diff
@@ -372,7 +372,7 @@ func (g *Globe) collectEnvironmentsInPath(searchPath string) []string {
 	return result
 }
 
-func (g Globe) isEnvPath(path string) bool {
+func (g *Globe) isEnvPath(path string) bool {
```
[the remark this answers is not in the document]
Comment (fritzduchardt) [reply internal/myks/globe.go:375]: Actually, I just got triggered by a warning in my IDE: "Struct Globe has both methods on pointer and value receivers. Such usage is not recommended by Go Documentation".

**Comment by fritzduchardt** [comment]
Hi @Zebradil , thanks for the review! Please have another look.

**Review by Zebradil — APPROVED** [review]

**Pull request merged**